### PR TITLE
stgit: Add completion variants for bash, zsh, fish

### PR DESCRIPTION
**Pull request by judaew**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.2.3 20D91
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/stgit/Portfile [hunk 1]
```diff
@@ -59,6 +60,35 @@ post-destroot {
     }
 }
 
+default_variants +bash_completion
+
+variant bash_completion {
```
Comment (cjones051073) [inline devel/stgit/Portfile:65]: Please address the lint errors by adding a variant description

File: devel/stgit/Portfile [hunk 2]
```diff
@@ -59,6 +60,35 @@ post-destroot {
...
+    post-destroot {
+        set bash_compl_path ${prefix}/share/bash-completion/completions
+        xinstall -d ${destroot}${bash_compl_path}
+        move ${destroot}${prefix}/share/stgit/completion/stgit.bash \
+                 ${destroot}${bash_compl_path}/stg
+    }
+}
+
+variant zsh_completion {
```
Comment (cjones051073) [inline devel/stgit/Portfile:74]: Requires a description

File: devel/stgit/Portfile [hunk 3]
```diff
@@ -59,6 +60,35 @@ post-destroot {
...
+    post-destroot {
+        set zsh_compl_path ${prefix}/share/zsh/site-functions
+        xinstall -d ${destroot}${zsh_compl_path}
+        move ${destroot}${prefix}/share/stgit/completion/stgit.zsh \
+                 ${destroot}${zsh_compl_path}/_stg
+    }
+}
+
+variant fish_completion {
```
Comment (cjones051073) [inline devel/stgit/Portfile:83]: Requires a description

**Review by cjones051073 — CHANGES_REQUESTED** [review]

**Comment by judaew** [comment]
Done

**Comment by judaew** [comment]
There is a global variable for `bash_completion` and `zsh_completion`:
https://github.com/macports/macports-ports/blob/master/_resources/port1.0/variant_descriptions.conf

For `fish_completion`, I've added a description.

**Review by cjones051073 — APPROVED** [review]

**Pull request merged**